src/main/java/io/cryostat/recordings/RecordingArchiveHelper.java

```diff
@@ -859,6 +860,26 @@ public Future<Path> getRecordingPath(String sourceTarget, String recordingName)
         return future;
     }
 
+    public byte[] getRecordingForDownload(String sourceTarget, String recordingName)
+            throws ExecutionException, RecordingNotFoundException {
+        byte[] recordingFile = null;
+        try {
+            Path recordingPath = getRecordingPath(sourceTarget, recordingName).get();
+            if (!env.hasEnv(Variables.DISABLE_ARCHIVE_COMPRESS) && isGzip(recordingPath)) {
+                return unGzip(recordingPath);
+            }
+            recordingFile = Files.readAllBytes(recordingPath);
```

Comment:
This will load the entire recording file into RAM within Cryostat's process, which is probably within a container with a relatively small memory allocation. If the requested recording file is close to that same size or larger then this operation will fail because there is not enough memory available to allocate. I think we need to find a way to return a Stream from this method and pump that Stream out to the HTTP response in the endpoint handler.

Reply:
I am trying to use the `pipeFromInput` as `TargetRecordingGetHandler` but it is not returning response and always returning an empty file. Would you mind looking and help me to find what is wrong?

Reply:
Sure, I'll see if I can find a bit of time to look into this tomorrow or Friday. If not then I am on PTO the week after, but I can take a look the week after that if you're still blocked.

Reply:
@lkonno I think I got it working, at least the immediate bug we were just talking about. I was able to download an archived recording through the web-client UI.

Reply:
Thanks!